### ISSUE TITLE
Getting it to work with pretty url's

### ISSUE THREAD

**Issue by jakoboo** [SmItH197/SteamAuthentication#115]
Am currently using [https://github.com/Teddy95/EasyRouter](url) to make pretty url's.
I had problem to even login 'cause it keeps getting stuck on /?login so i've changed this lines:
```php
if(!$openid->mode) {
    //if(isset($_GET)) {
    if($_GET['page'] == "?login") {
        $openid->identity = 'http://steamcommunity.com/openid';
        header('Location: ' . $openid->authUrl());
    }
    echo "<form action=\"?login\" method=\"post\"> <input type=\"image\" src=\"http://cdn.steamcommunity.com/public/images/signinthroughsteam/sits_".$button.".png\"></form>";
}
````

But now it keeps getting stuck on this after login on steam
```php
function __get($name)
{
    switch ($name) {
    case 'identity':
        # We return claimed_id instead of identity,
        # because the developer should see the claimed identifier,
        # i.e. what he set as identity, not the op-local identifier (which is what we verify)
        return $this->claimed_id;
    case 'trustRoot':
    case 'realm':
        return $this->trustRoot;
    case 'mode':
        /*----------here----------*/
        /*$this->data['openid_mode'] IS EMPTY here but anywhere else it returns correct value*/
        return empty($this->data['openid_mode']) ? null : $this->data['openid_mode'];
    }
}
```

my `$this->data` is
`$this->data = ($_SERVER['REQUEST_METHOD'] === 'POST') ? $_POST : $GLOBALS["true_get_params"];`

according to EasyRouter doc
`$true_get_params = EasyRouter\main::get_true_params();`

Any way to get it to work with pretty url's? Maybe I should change Teddy95/EasyRouter to something different?

**Comment by bman46** [comment]
Why do you need the pretty url?

**Comment by jakoboo** [comment]
emm to have pretty url's? like /user instead of /user.php etc

**Comment by bman46** [comment]
Oh k

**Comment by bman46** [comment]
can u try it with out the variable changing thing?

**Comment by jakoboo** [comment]
i've get it to work by using https://github.com/mrjgreen/phroute instead

**Comment by bman46** [comment]
K

**Comment by SmItH197** [comment]
Closed, but for future reference see #131.
